### PR TITLE
Add kubevirt_vm_resource_requests metric for memory resource

### PR DESCRIPTION
**Pull request by machadovilaca**
### What this PR does
Before this PR:

No metric with VM resource requests data

After this PR:

`kubevirt_vm_resource_requests` hold memory resource requests for VMs

jira-ticket: https://issues.redhat.com/browse/CNV-44275

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
Add kubevirt_vm_resource_requests metric for memory resource
```

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -93,6 +93,9 @@ Virtual Machine last transition timestamp to migrating status. Type: Counter.
 ### kubevirt_vm_non_running_status_last_transition_timestamp_seconds
 Virtual Machine last transition timestamp to paused/stopped status. Type: Counter.
 
+### kubevirt_vm_resource_requests
+Resources requested by Virtual Machine. Type: Gauge.
```
Comment (sradco) [inline docs/metrics.md:97]: @machadovilaca  Can you please mention these are CPU and memory requests?

Comment (machadovilaca) [reply docs/metrics.md:97]: cpu will only be added in https://github.com/kubevirt/kubevirt/pull/12625

**Comment by sradco** [comment]
@machadovilaca Please update pr name

**Pull request merged**